src/cps/pc_compat/pc_loader.cpp

```diff
@@ -0,0 +1,168 @@
+        return ost;
+    }
+
+    PcLoader::PcLoader() = default;
+
+    tl::expected<loader::Package, std::string> PcLoader::load(std::istream & istream, fs::path const & filename) {
+        scan_begin(istream);
+        yy::parser parse(*this);
+        // To debug parser, uncomment the following line
+        // TODO: add a way to enable debug output without rebuilding
+        // parse.set_debug_level(true);
+        if (const int result = parse(); result != 0) {
+            throw std::runtime_error("Failed to parse the given pkg-config file.");
+        }
+
+        std::string name = CPS_TRY(get_property("Name").and_then(get_string));
+
+        loader::LangValues compile_flags;
+        if (auto compile_flags_input = get_property("Cflags").and_then(get_string)) {
+            auto compile_flags_vec = utils::split(*compile_flags_input);
+            compile_flags.emplace(loader::KnownLanguages::c, compile_flags_vec);
+            compile_flags.emplace(loader::KnownLanguages::cxx, compile_flags_vec);
+            compile_flags.emplace(loader::KnownLanguages::fortran, compile_flags_vec);
+        }
+
+        std::vector<std::string> link_flags;
+        if (auto link_flags_input = get_property("Libs").and_then(get_string)) {
+            link_flags = cps::utils::split(*link_flags_input);
+        }
+
+        std::vector<std::string> require;
+        if (auto requires_input = get_property("Requires").and_then(get_package_requirements)) {
+            std::transform(requires_input->begin(), requires_input->end(), std::back_inserter(require),
+                           [](const PackageRequirement & requirement) { return requirement.package; });
+        }
+
+        std::unordered_map<std::string, loader::Component> components;
+        components.emplace(
+            name, loader::Component{.type = loader::Type::unknown,
+                                    .compile_flags = compile_flags,
+                                    .includes = loader::LangValues{},
+                                    .definitions = loader::Defines{},
+                                    .link_flags = link_flags,
+                                    .link_libraries = {},
+                                    // TODO: Currently lib location is hard coded to appease assertions. This would
+                                    // need to implement linker-like search to replicate current behavior.
+                                    .location = fmt::format("@prefix@/lib/{}.a", name),
+                                    .link_location = std::nullopt,
+                                    .require = require});
+
+        return loader::Package{.name = name,
+                               .cps_version = std::string{loader::CPS_VERSION},
+                               .components = components,
+                               // TODO: treat PREFIX in pc file specially and translate it to @prefix@
+                               .cps_path = std::nullopt,
```

Comment:
We *do* know the pc file path. I feel like we should model that to assist in debugging.

Options:

* Make an issue, land this PR, and come back to that
* Use `cps_path` and point at a `*.pc` file and expect we'll figure it out
* Add an extension [1] field for pc_path.

[1] CPS doesn't specify how to add a nonstandard extension, but we could just make something up and we'll clean it up later.

Reply:
Good point. Did that and added a TODO to refactor the interface of the `load` function. Currently `filename` is actually the parent directory path.

Reply:
No actually, I totally forgot why I left that TODO in there. Prefix calculating logic on the CPS side is currently expecting something like this: https://github.com/cps-org/cps-config/blob/607c44900004f382271a0c1cfdd5dfe3a7977384/tests/cps-files/lib/cps/cps-path-set.cps.in#L4. So yeah, setting it to null works best for this PR, I think. Let's add a follow up issue to figure out how to calculate prefix for pc files